tests/test8.in

```diff
@@ -4,11 +4,18 @@ logmsg Test variable expand
 set n 10
 set test (seq $n)
 echo $test[1..$n]      # normal range
+echo $test[ .. ]       # implied range from 1 to $n
+echo $test[ ..3]       # range from implied 1
+echo $test[8.. ]       # range up to implied $n
 echo $test[$n..1]      # inverted range
 echo $test[2..5 8..6]  # several ranges
 echo $test[-1..-2]     # range with negative limits
 echo $test[-1..1]      # range with mixed limits
 
+# missing start, cannot use implied range
+echo $test[1..2 ..]
+echo $test[..1 ..2]
+
```

Comment:
Please add a test that makes the behavior clear for an empty variable, e.g.:

    echo $test[..$empty]

I do not have a strong opinion about what that should do, but it should be something we decide and test.

Reply:
Ah right, I added a test for it.
Turns out this expands to no indices because of cartesian product expansion. It is at least consistent with the same code with 1 on the left side.